span.go

```diff
@@ -318,3 +318,8 @@ func setSamplingPriority(s *Span, value interface{}) bool {
 	}
 	return false
 }
+
+// SetFirehose sets the firehose tag on the span context
+func SetFirehose(s *Span) {
```

Comment:
EnableFirehose might be better name, as `set()` usually implies a value argument